crates/precompile/src/interface.rs

```diff
@@ -16,12 +16,33 @@ pub struct PrecompileOutput {
     pub gas_used: u64,
     /// Output bytes
     pub bytes: Bytes,
+    /// Whether the precompile reverted
+    pub reverted: bool,
```

Comment:
this seems like the least invasive option to me,

the use case here is valid, since there exist precompiles that act as revert